### ISSUE TITLE
Invalid file descriptor to ICU data received

### ISSUE THREAD

**Issue by bascurtiz** [pyinstaller/pyinstaller#5400]
## Description of the issue

When running the following code in main.py:

  
    webview.create_window('Example', 'https://pywebview.flowrl.com/hello')
    webview.start(debug=True, gui='cef')

....All is good and shows the site in a window.

After using PyInstaller 4.1 wth `pyinstaller --onefile main.spec` it ends with Building EXE completed successfully.
So I open up \dist\main.exe which confronts me with:
`[ERROR:icu_util.cc(133)] Invalid file descriptor to ICU data received`

When I replace the line in main.py with `webview.start(debug=True)` all is good.
But I need CEF in order to show right-click menu and better compatibility.

I simply installed cefpython3 with `pip install cefpython3` which didn't show up any errors during install.
I noticed https://github.com/cztomczak/cefpython/tree/master/examples/pyinstaller , but those are for Linux.

## Context information (for bug reports)

* Output of `pyinstaller --version`: ```4.1```
* Version of Python:  ```3.6.8```
* Platform: Windows 10 (English) 2004

main.spec code:
```
# -*- mode: python ; coding: utf-8 -*-

block_cipher = None


a = Analysis(['main.py'],
             pathex=[''],
             binaries=[],
             datas=[('', '.')],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas,
          [],
          name='main',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          upx_exclude=[],
          runtime_tmpdir=None,
          console=True )
```

**Comment by bwoodsend** [comment]
>  I noticed [https://github.com/cztomczak/cefpython/tree/master/examples/pyinstaller]() , but those are for Linux.

No, if you look at the hook itself you'll see plenty of `is_win32` and `is_darwin` code so it's definitely not just for Linux. I'm guessing that all you need to do is download that hook, put it in the same folder as `main.spec`, add `'.'` to `hookspath=[],` in the spec, then rebuild using the `--clean` option.

**Comment by bascurtiz** [comment]
@bwoodsend 
Thanks!
I did as you described, downloaded the `hook-cefpython3.py`, made sure it's in same folder as `main.spec`, added replaced hookspath with 'hookspath='.'` in the spec, then rebuild with `pyinstaller --onefile --clean main.spec`

Which results in the following error:
```
PyInstaller.exceptions.ImportErrorWhenRunningHook: 
Failed to import module __PyInstaller_hooks_0_cefpython3 required by hook for module C:\Users\useraccount\Downloads\boiler\hook-cefpython3.py. 
Please check whether module __PyInstaller_hooks_0_cefpython3 actually exists and whether the hook is compatible with your version of C:\Users\useraccount\Downloads\boiler\hook-cefpython3.py: 
You might want to read more about hooks in the manual and provide a pull-request to improve PyInstaller.
```
When googling I found this solution, [here](https://stackoverflow.com/a/41214596), but despite following this up, I keep getting the same error.

Mind you, `pyinstaller --version` now shows `4.2.dev0+2771f6aff5`

When searching in `C:\Users\useraccount\AppData\Local\Programs\Python\Python36\Lib\site-packages\pyinstaller-4.2.dev0+2771f6aff5-py3.6.egg` there is indeed no such file called `__PyInstaller_hooks_0_cefpython3`.

In `C:\Users\useraccount\AppData\Local\Programs\Python\Python36\Lib\site-packages\cefpython3\examples\pyinstaller\__pycache__` there is `hook-cefpython3.cpython-36.py` but not sure if it's refering to that file.

**Comment by rokm** [comment]
That hook will probably need to be adjusted for the latest version(s) of PyInstaller.

If nothing else, `is_py2` does not exist anymore:
```python
from PyInstaller.compat import is_win, is_darwin, is_linux, is_py2
```

**Comment by bwoodsend** [comment]
Ugh, actually on looking at the rest of the contents of that directory I see this is quite messy. It looks like they've tried to write their own PyInstaller support and tied themselves in a knot doing it. I'll write a proper hook for it. 

> webview.create_window('Example', 'https://pywebview.flowrl.com/hello')
> webview.start(debug=True, gui='cef')

Can you expand this into a self-contained example I can use as a test? i.e. Define `webview` so I can run it directly. I have no idea what these libraries are...

**Comment by bascurtiz** [comment]
@bwoodsend 
Thank you!

Full code of `main.py` below:
```
import webview
from cefpython3 import cefpython as cef

if __name__ == '__main__':
    webview.create_window('Example', 'https://pywebview.flowrl.com/hello')
    webview.start(debug=True, gui='cef')
```

**Comment by rokm** [comment]
@bascurtiz modifying the hook and replacing
```python
from PyInstaller.compat import is_win, is_darwin, is_linux, is_py2
```
with 
```python
from PyInstaller.compat import is_win, is_darwin, is_linux
is_py2 = False
```
should do the trick.

**Comment by bascurtiz** [comment]
@rokm 
Thank you, but this results in:
`  
File "C:\Users\useraccount\Downloads\boiler\hook-cefpython3.py", line 17
    from PyInstaller.compat import is_win, is_darwin, is_linux, is_py2 = False
                                                                       ^
SyntaxError: invalid syntax
`
Also tried  `from PyInstaller.compat import is_win, is_darwin, is_linux, is_py2=False`
Same result.

**Comment by rokm** [comment]
```python
from PyInstaller.compat import is_win, is_darwin, is_linux
# is_py2 goes in separate line
is_py2 = False
```

**Comment by bascurtiz** [comment]
Oops! Will try asap and let you know.

**Comment by bascurtiz** [comment]
@rokm 
Works \o/

**Comment by bwoodsend** [comment]
Excellent. @bascurtiz I suggest you open a pull request to cefpython to send them your updated version. If you're feeling keen, [PyInstaller 4.0 introduced a way for libraries to declare their own hooks](https://pyinstaller.readthedocs.io/en/stable/hooks.html#provide-hooks-with-package) so if you followed the steps in there and include that as part of your pull request, then you won't have to copy the hook manually for future versions of cefpython - it'll be found automatically.

**Comment by bascurtiz** [comment]
@bwoodsend 
Uhm, I'm just a noobie, so I guess it's best you pick this up.
I got it working for myself thanks to @rokm , but for future projects and other users, it might be handy if you do.
...or that I make an issue on their repo to refer to this one?

**Comment by bwoodsend** [comment]
>  Uhm, I'm just a noobie, so I guess it's best you pick this up.

We all are until we take the plunge and submit our first pull request. But ok, I'll do it then...

**Comment by rokm** [comment]
Closing this as `cefpython3` [fixed their hook for PyInstaller >= 4.0 compatibility](https://github.com/cztomczak/cefpython/commit/686d528b3eebbe5b84a770f5d266aea8dd185721).

**Comment by baight** [comment]
I got the same issue. I solved it by copying "icudtl.dat、locales、cef.pak、cef_100_percent.pak……" into the .exe directory.

we can use 'pip show cefpython3' to find out where the files are. For example, my cefpython3 path is "c:\users\baight\appdata\local\programs\python\python38\lib\site-packages".  So we can find the files in "c:\users\baight\appdata\local\programs\python\python38\lib\site-packages\cefpython3".

I am not sure which files should copy, so I copied all of them, except "__pycache__"、"examples"、"swiftshader"、"__init__.py"、"debug.log"、"License"、"README.txt"……, that obviously not needed.
